src/rust/engine/process_executor/src/main.rs

```diff
@@ -459,14 +332,224 @@ async fn main() {
   exit(result.exit_code);
 }
 
-fn collection_from_keyvalues<'a, It, Col>(keyvalues: It) -> Col
+async fn make_request(
+  store: &Store,
+  args: &Opt,
+) -> Result<(process_execution::Process, ProcessMetadata), String> {
+  match (
```

Comment:
These match arms have a lot of code in them. Probably worth splitting out some methods to simplify them.